src/components/views/location/LocationShareMenu.tsx

```diff
@@ -29,22 +31,40 @@ type Props = Omit<ILocationPickerProps, 'onChoose'> & {
     roomId: Room["roomId"];
 };
 
+const useEnabledShareTypes = (): LocationShareType[] => {
+    const isPinDropLocationShareEnabled = SettingsStore.getValue("feature_location_share_pin_drop");
+
+    if (isPinDropLocationShareEnabled) {
+        return [LocationShareType.Own, LocationShareType.Pin];
+    }
+    return [
+        LocationShareType.Own,
+    ];
+};
```

Comment:
This will be evaluated as a hook due to its name beginning with `use` yet it composes no hooks